src/ngraph_encapsulate_clusters.cc

```diff
@@ -256,6 +267,37 @@ Status EncapsulateClusters(Graph* graph) {
       cluster_input_map[dst_cluster_idx].push_back(
           std::make_tuple(src->id(), edge->src_output(), dt));
     }
+
+    if (config::IsLoggingPlacement()) {
+      if (edge_is_arg && edge_is_retval) {
+        count_both_arg_retval++;
+      } else {
+        if (edge_is_arg) {
+          count_arg++;
+        } else {
+          count_retval++;
+        }
+      }
+    }
+  }
+
+  if (config::IsLoggingPlacement()) {
+    int computed_edge_number = count_arg + count_retval +
+                               count_both_arg_retval + count_free +
+                               count_encapsulated;
+    std::cout << "NGTF_SUMMARY: Types of edges:: args: " << count_arg
+              << ", retvals: " << count_retval
+              << ", both arg and retval: " << count_both_arg_retval
+              << ", free: " << count_free
+              << ", encapsulated: " << count_encapsulated
+              << ", total: " << count_tot
+              << ", computed total: " << computed_edge_number << endl;
+    if (!(computed_edge_number == count_tot &&
+          count_tot == graph->num_edges())) {
+      return errors::Internal("Computed number of edges ", computed_edge_number,
```

Comment:
Please separate out the check as the error message does not clarify when count_tot!=graph->num_edges

Reply:
This assert is trying to check a==b==c. So printing out all 3 numbers when the check fails